### ISSUE TITLE
Ability to configure kill timeout for processes

### ISSUE THREAD

**Issue by DamonOehlman** [buildkite/agent#393]
Hey folks,

Just wondering if the ability to configure the kill timeout for an agent might be on the roadmap anywhere?  I can see the code where it is executed (in `process.go`):

``` go
    // Timeout this process after 3 seconds
    select {
    case _ = <-c:
        // Was successfully terminated
    case <-time.After(10 * time.Second):
        // Stop checking in the routine above
        checking = false

        // Forcefully kill the thing
        err = p.signal(syscall.SIGKILL)

        if err != nil {
            return err
        }
    }
```

If it isn't at this stage, I might look at trying to get my go on at some stage and submitting a PR.

Cheers,
Damon.

**Comment by keithpitt** [comment]
@DamonOehlman not on the road map yet, but if you submit a PR for it, we'd happily merge it and push out a stable :)

It's always been on the cards as something we want to add.

Only tricky things I can think of are:

- What's the config value called?
- How do you configure it?

The 10 seconds there was just some arbitrary number I chose - no science was behind it at all! So having it configuring would be awesome.

**Comment by JensRantil** [comment]
There's a command line utility called `timeout` that you could wrap your commands in as a workaround.

**Comment by lox** [comment]
Closed in #700!
